src/realm-settings/key-providers/aes-generated/AESGeneratedForm.tsx

```diff
@@ -0,0 +1,360 @@
+        labelIcon={
+          <HelpItem
+            helpText="realm-settings-help:active"
+            forLabel={t("active")}
+            forID="kc-active"
+          />
+        }
+      >
+        <Controller
+          name="config.active"
+          control={form.control}
+          defaultValue={["true"]}
+          render={({ onChange, value }) => {
+            return (
+              <Switch
+                id="kc-active"
+                label={t("common:on")}
+                labelOff={t("common:off")}
+                isChecked={value[0] === "true"}
+                data-testid={
+                  value[0] === "true"
+                    ? "internationalization-enabled"
+                    : "internationalization-disabled"
+                }
+                onChange={(value) => {
+                  onChange([value.toString()]);
+                }}
+              />
+            );
+          }}
+        />
+      </FormGroup>
+      <FormGroup
+        label={t("AESKeySize")}
+        fieldId="kc-aes-keysize"
+        labelIcon={
+          <HelpItem
+            helpText="realm-settings-help:AESKeySize"
+            forLabel={t("AESKeySize")}
+            forID="kc-aes-key-size"
+          />
+        }
+      >
+        <Controller
+          name="config.secretSize"
+          control={form.control}
+          defaultValue={["16"]}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-aes-keysize"
+              onToggle={() => setIsKeySizeDropdownOpen(!isKeySizeDropdownOpen)}
+              onSelect={(_, value) => {
+                onChange([value.toString()]);
+                setIsKeySizeDropdownOpen(false);
+              }}
+              selections={[value.toString()]}
+              isOpen={isKeySizeDropdownOpen}
+              variant={SelectVariant.single}
+              aria-label={t("aesKeySize")}
+              data-testid="select-secret-size"
+            >
+              {aesSecretSizeOptions?.map((item) => (
+                <SelectOption
+                  selected={item === value}
+                  key={item}
+                  value={item}
+                />
+              ))}
+            </Select>
+          )}
+        />
+      </FormGroup>
+      <ActionGroup className="kc-AESform-buttons">
+        <Button
+          className="kc-AESform-save-button"
+          data-testid="add-provider-button"
+          variant="primary"
+          type="submit"
+        >
+          {t("common:save")}
+        </Button>
+        <Button
+          className="kc-AESform-cancel-button"
+          onClick={(!editMode && handleModalToggle) || undefined}
+          variant="link"
+        >
+          {t("common:cancel")}
+        </Button>
+      </ActionGroup>
+    </FormAccess>
+  );
+};
+
+export const AESGeneratedSettings = () => {
+  const { t } = useTranslation("realm-settings");
+  const providerId = useRouteMatch<MatchParams>(AESProviderTypeRoute.path)
```

Comment:
Since this match is using the path from `AESProviderTypeRoute` we should also use the corresponding type from that file.

```suggestion
  const providerId = useRouteMatch<AESProviderTypeRouteParams>(AESProviderTypeRoute.path)
```

But since this should be the only route that is triggered anyways I believe we can also use `useParams` here.
```suggestion
  const params = useParams<AESProviderTypeRouteParams>()
```